app/screens/home/index.tsx

```diff
@@ -56,7 +56,17 @@ export default function HomeScreen(props: HomeProps) {
         return () => {
             listener.remove();
         };
-    });
+    }, [intl]);
+
+    useEffect(() => {
+        const listener = DeviceEventEmitter.addListener(Events.LEAVE_CHANNEL, (displayName: string) => {
+            alertChannelRemove(displayName, intl);
+        });
+
+        return () => {
+            listener.remove();
+        };
+    }, [intl]);
```

Comment:
```suggestion
    }, [intl.locale]);
```

is this better than checking for the entire object?

Reply:
Haven't done any proper checks on this, but...

If intl is properly memoized, we are storing one object reference (a word (4 bytes) I guess). If we store a string, it is at least two chars (2 bytes?) (or even 5 if it is something like en-CA). So memory-wise is more stable to store the object. And since we are using it in other places (there are other places we call intl.formatMessage) I guess the object stays in the heap either way.

In terms of performance, we would be talking about comparing a reference (comparing one word, which should be O(1)), and the other place is comparing a string, that is in the worst case (unless javascript has underlying optimizations) O(n).

So... in theory, should be better to check the object (if it is properly memoized and doesn't change as long the locale/formatMessage doesn't change), but the gain is probably neglectable.